### PR TITLE
Added first network call.

### PR DESCRIPTION
**Pull request by PabloGrelaM**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 Project to learn about architecture and design patterns in Android.
 
-You will need an API key from https://superheroapi.com/ to run this project.
+We didn't find a proper super hero API, so we decided to go with Rick & Morty!
```
Comment (andreslopezsarmiento) [inline README.md:5]: LOL

File: app/src/main/java/com/mstudio/superheroarch/MainActivity.kt [hunk 2]
```diff
@@ -18,8 +26,29 @@ class MainActivity : AppCompatActivity() {
...
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
         findViewById<Button>(R.id.mainButton).setOnClickListener {
-            findViewById<TextView>(R.id.mainTextView).text = getString(R.string.changed_string)
+            CoroutineScope(Dispatchers.IO).launch {
+                val response = retrofit.create(RickAndMortyApiService::class.java).getCharacters()
+
+                if (response.isSuccessful) {
+                    val characterList = response.body()
+                    runOnUiThread {
```
Comment (andreslopezsarmiento) [inline app/src/main/java/com/mstudio/superheroarch/MainActivity.kt:40]: why is this needed here and not in the error case where you show a snackbar? Are snackbars thread-safe?

Comment (PabloGrelaM) [reply app/src/main/java/com/mstudio/superheroarch/MainActivity.kt:40]: seems like 😄
